File: Leetcode/14-Longest-Common-Prefix/LongestCommonPrefix.go
func longestCommonPrefix(strs []string) string {
	if strs == nil || len(strs) == 0 {
		return ""
	}
	if len(strs) == 1 {
		return strs[0]
	}
	for i := 0; i < len(strs[0]); i++ {
		ch := strs[0][i]
		for _, str := range strs {
			if i < len(str) && ch == str[i] {
				continue
			}
			return str[0:i]
		}
	}
	return strs[0]
}

File: Leetcode/14-Longest-Common-Prefix/LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) {
            return "";
        }
        if (strs.size() == 1) {
            return strs[0];
        }
        for (int i=0; i<strs[0].length(); i++) {
            char c = strs[0].at(i);
            for (int j=1; j<strs.size(); j++) {
                if (i<strs[j].length() && c==strs[j].at(i)) {
                    continue;
                }
                return strs[0].substr(0,i);
            }
        }
        // strs[0] could be the shortest.
        return strs[0];
    }
};
